lib/phone/nanp.ex

```diff
@@ -1,32 +1,85 @@
 defmodule Phone.NANP do
-  use Helper.Country
-  field :modules, [
-    Phone.NANP.AS,
-    Phone.NANP.AI,
-    Phone.NANP.AG,
-    Phone.NANP.BS,
-    Phone.NANP.BB,
-    Phone.NANP.BM,
-    Phone.NANP.CA,
-    Phone.NANP.DM,
-    Phone.NANP.DO,
-    Phone.NANP.GD,
-    Phone.NANP.GU,
-    Phone.NANP.JM,
-    Phone.NANP.KN,
-    Phone.NANP.KY,
-    Phone.NANP.LC,
-    Phone.NANP.MP,
-    Phone.NANP.MS,
-    Phone.NANP.PR,
-    Phone.NANP.SX,
-    Phone.NANP.TC,
-    Phone.NANP.TT,
-    Phone.NANP.US,
-    Phone.NANP.VC,
-    Phone.NANP.VG,
-    Phone.NANP.VI,
-    Phone.NANP.TollFree
-  ]
-  match :modules
+  def build("1684" <> _ = number), do: Phone.NANP.AS.build(number)
```

Comment:
It's cool to see the performance improvement for these, but it's a bummer that these prefixes have to be duplicated everywhere. Could the `build/1` and `build!/1` functions be written using some sort of macro?